pkg/util/grpc.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc"
+)
+
+func ServerStreamWithContext(ctx context.Context, stream grpc.ServerStream) grpc.ServerStream {
+	return &serverStreamWithContext{
+		ServerStream: stream,
+		ctx:          ctx,
+	}
+}
+
+type serverStreamWithContext struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+var _ grpc.ServerStream = (*serverStreamWithContext)(nil)
+
+func (s *serverStreamWithContext) Context() context.Context {
+	return s.ctx
+}
+
+// Split '/pkg.Service/Method' into 'pkg.Service' and 'Method'
+func SplitFullyQualifiedMethodName(fqn string) (string, string, bool) {
+	parts := strings.Split(fqn, "/")
```

Comment:
nit
```suggestion
	parts := strings.SplitN(fqn, "/", 4)
```

You could also assume that the fully qualified method name is valid and return `Cut(TrimPrefix(fqn, "/"), "/")`, which will result in a fullname for `/a/b/c` of `a.b/c` (when looking up the method in the global registry) which is just not gonna be found.